pyaedt/edb_grpc/core/edb_data/padstacks_data.py

```diff
@@ -0,0 +1,1656 @@
+                    via._edb_padstackinstance.GetNet(),
+                    generate_unique_name(instance.GetName()),
+                    instance,
+                    via._edb_padstackinstance.GetPositionAndRotationValue()[1],
+                    via._edb_padstackinstance.GetPositionAndRotationValue()[2],
+                    from_layer,
+                    to_layer,
+                    None,
+                    None,
+                )
+                padstack_instance.SetIsLayoutPin(via.is_pin)
+                i += 1
+            via.delete()
+        self._ppadstack._pedb.logger.info("Created {} new microvias.".format(i))
+        return new_instances
+
+
+class EDBPadstackInstance(object):
+    """Manages EDB functionalities for a padstack.
+
+    Parameters
+    ----------
+    edb_padstackinstance :
+
+    _pedb :
+        Inherited AEDT object.
+
+    Examples
+    --------
+    >>> from pyaedt import Edb
+    >>> edb = Edb(myedb, edbversion="2021.2")
+    >>> edb_padstack_instance = edb.core_padstack.instances[0]
+    """
+
+    def __getattr__(self, key):
+        try:
+            return self[key]
+        except:
+            try:
+                return getattr(self._edb_padstackinstance, key)
+            except AttributeError:
+                raise AttributeError("Attribute not present")
+
+    def __init__(self, edb_padstackinstance, _pedb):
+        self._edb_padstackinstance = edb_padstackinstance
+        self._pedb = _pedb
+        self._bounding_box = []
+        self._object_instance = None
+        self._position = []
+        self._pdef = None
+
+    @property
+    def object_instance(self):
+        """Edb Object Instance."""
+        if not self._object_instance:
+            self._object_instance = (
+                self._edb_padstackinstance.GetLayout()
+                .GetLayoutInstance()
+                .GetLayoutObjInstance(self._edb_padstackinstance, None)
+            )
+        return self._object_instance
+
+    @property
+    def bounding_box(self):
+        """Get bounding box of the padstack instance.
+        Because this method is slow, the bounding box is stored in a variable and reused.
+
+        Returns
+        -------
+        list of float
+        """
+        if self._bounding_box:
+            return self._bounding_box
+        bbox = self.object_instance.GetBBox()
+        self._bounding_box = [
+            [bbox.Item1.X.ToDouble(), bbox.Item1.Y.ToDouble()],
+            [bbox.Item2.X.ToDouble(), bbox.Item2.Y.ToDouble()],
+        ]
+        return self._bounding_box
+
+    @pyaedt_function_handler()
+    def in_polygon(self, polygon_data, include_partial=True, simple_check=False):
+        """Check if padstack Instance is in given polygon data.
+
+        Parameters
+        ----------
+        polygon_data : PolygonData Object
+        include_partial : bool, optional
+            Whether to include partial intersecting instances. The default is ``True``.
+
```

Comment:
```suggestion
        """Check if padstack instance is in given polygon data.

        Parameters
        ----------
        polygon_data : PolygonData Object
        include_partial : bool, optional
            Whether to include partial intersecting instances. The default is ``True``.
        simple_check : bool, optional

```
Add descriptions for polygon_data and simple_check.